docs/user/ppl/admin/datasources.rst

```diff
@@ -25,44 +25,91 @@ Definitions of datasource and connector
 
 Example Prometheus Datasource Definition ::
 
-    [{
+    {
         "name" : "my_prometheus",
         "connector": "prometheus",
         "properties" : {
             "prometheus.uri" : "http://localhost:8080",
             "prometheus.auth.type" : "basicauth",
             "prometheus.auth.username" : "admin",
             "prometheus.auth.password" : "admin"
-        }
-    }]
+        },
+        "allowedRoles" : ["prometheus_access"]
+    }
 Datasource configuration Restrictions.
 
 * ``name``, ``connector``, ``properties`` are required fields in the datasource configuration.
+* In case of secure domains, ``allowedRoles`` can be used to specify the opensearch roles allowed to access the datasource via PPL/SQL.
```

Comment:
1. allowedRoles is optional or mandatory?
2. is prometheus_access predefined role? where is prometheus_access defined?

Reply:
prometheus_access is not a predefined role. user can define any role and add it to the datasource.
Domains with security plugin:
Not mandatory, it will imply only admin can access the datasource.
Domains without security plugin:
Not mandatory, anyone can access datasource.

Reply:
```
Datasource configuration Restrictions.

* ``name``, ``connector``, ``properties`` are required fields in the datasource configuration.
* In case of secure domains, ``allowedRoles`` can be used to specify the opensearch roles allowed to access the datasource via PPL/SQL.
* If ``allowedRoles`` are not specified for a datasource, only users with ``all_access`` could access the datasource in case of secure domains.
* In case of security disabled domains, authorization is disbaled.
* All the datasource names should be unique and match the following regex[``[@*A-Za-z]+?[*a-zA-Z_\-0-9]*``].
* Allowed Connectors.
    * ``prometheus`` [More details: `Prometheus Connector <prometheus_connector.rst>`_]
* All the allowed config parameters in ``properties`` are defined in individual connector pages mentioned above.
```

Reply:
Changed to above text. Let me know if it is ok.